airbyte-integrations/connectors/source-amazon-ads/unit_tests/test_report_streams.py

```diff
@@ -136,6 +137,59 @@ def test_display_report_stream(config):
     assert len(metrics) == METRICS_COUNT * (len(stream.metrics_map) - 1)
 
 
+@pytest.mark.parametrize(
+    ("profiles", "stream_class", "url_pattern", "expected"),
+    [
+        (
+            make_profiles(),
+            SponsoredDisplayReportStream,
+            r"https://advertising-api.amazon.com/sd/([a-zA-Z]+)/report",
+            True,
+        ),
+        (
+            make_profiles(profile_type="vendor"),
+            SponsoredDisplayReportStream,
+            r"https://advertising-api.amazon.com/sd/([a-zA-Z]+)/report",
+            False,
+        ),
+        (
+            make_profiles(),
+            SponsoredProductsReportStream,
+            r"https://advertising-api.amazon.com/v2/sp/([a-zA-Z]+)/report",
+            True,
+        ),
+        (
+            make_profiles(profile_type="vendor"),
+            SponsoredProductsReportStream,
+            r"https://advertising-api.amazon.com/v2/sp/([a-zA-Z]+)/report",
+            False,
+        ),
+    ],
+)
+@responses.activate
+def test_stream_report_body_metrics(config, profiles, stream_class, url_pattern, expected):
+    setup_responses(
+        init_response=REPORT_INIT_RESPONSE,
+        init_response_products=REPORT_INIT_RESPONSE,
+        status_response=REPORT_STATUS_RESPONSE,
+        metric_response=METRIC_RESPONSE,
+    )
+
+    stream = stream_class(config, profiles, authenticator=mock.MagicMock())
+    stream_slice = {"profile": profiles[0], "reportDate": "20210725"}
+    list(stream.read_records(SyncMode.incremental, stream_slice=stream_slice))
+    for call in responses.calls:
+        create_report_pattern = re.compile(url_pattern)
+        for match in create_report_pattern.finditer(call.request.url):
+            record_type = match.group(1)
+            request_body = call.request.body
+            request_metrics = json.loads(request_body.decode('utf-8'))['metrics']
+            if record_type == "productAds" or record_type == "asins":
```

Comment:
```suggestion
            if record_type == RecordType.PRODUCTADS or record_type == RecordType.ASINS:
```

Looks good overall. It seems like this is the last of my change request.

After it, please run `./gradlew format` in the Airbyte root directory to fix all formatting issues and commit all the changes it will make in the `source-amazon-ads` directory. I see some formatting issues in your code, for example, a single comma. This tool will fix them.

Reply:
@monai - Thank you for your feedback. I have made the changes. Please review.